src/main/java/com/miraisolutions/xlconnect/Workbook.java

```diff
@@ -354,22 +375,24 @@ public void createName(String name, String formula, boolean overwrite) {
     }
 
     public void removeName(String name) {
+        if (name == null)
+                return;
         Name cname = workbook.getName(name);
         if(cname != null)
             workbook.removeName(cname);
     }
```

Comment:
We need worksheetScope here too (see comment above). I am also not sure about the check for `name == null`, which we don't have anywhere. We will need to go through an `existsName()` as our `getName()` errors as opposed to `workbook.getName()`, sometheing like
```suggestion
    public void removeName(String name, String worksheetScope) {
        if (existsName(worksheetScope, name)) {
            Name cname = getNameInWorksheetScope(worksheetScope, name);
            workbook.removeName(cname);
        }
    }
```